packages/web/hooks/useGetWindowSize.ts

```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect } from 'react'
+
+/**
+ * A simple hook to get dimensions of window.
+ * Handles tracking and returning updated values on resizing.
+ *
+ * Example Usage:
+ * const windowSize = useGetWindowSize()
+ *
+ * if (windowSize.width === someBreakpoint) {
+ *  // Show mobile view
+ * }
+ */
+
+type WindowSize = {
+  width: number | null
+  height: number | null
+}
+
+const useWindowSize = () => {
+  // Make sure we're client-side / not server-side
+  const isClient = typeof window === 'object'
+
+  const [windowSize, setWindowSize] = useState<WindowSize>({
```

Comment:
I totally see the logic here, but this is one of those reference equality gotchas. Because you're passing an object literal to `useState` its value will be updated every time the hook is called (because `{} !== {}`, `useState` will see a new value each time it's called) so this will produce unnecessary re-rendering.

The most straight forward  approach would probably be to track window width and height as two different pieces of state, then at the bottom of the hook do `return useMemo(() => { width: windowWidth, height: windowHeight }, [windowWidth, windowHeight])` (where `windowWidth` and `windowHeight` are the two state values).